src/containers/Layers.tsx

```diff
@@ -60,10 +65,41 @@ class Layers extends React.Component<LayersProps> {
       },
       seedZones: { name: t`Seed Zones`, urlIdentifier: 'seedzones', awayMessage: null },
       layers: { name: t`Layers`, urlIdentifier: 'layers', awayMessage: null },
+      custom: {
+        name: t`Custom`,
+        urlIdentifier: 'custom',
+        awayMessage: null,
+      },
     }
 
-    const layerList = (urlIdentifier: string) =>
-      layers
+    const layerList = (urlIdentifier: string) => {
+      if (urlIdentifier === 'custom') {
+        return [
+          <div className="layer-list" key="shapeUpload">
+            <ShapefileUpload storeTo="customLayers">
+              <div className="is-clickable">
+                <span
+                  style={{
+                    margin: '2px',
+                    padding: '0px 4px 2px 4px',
+                    borderRadius: '100%',
+                    border: '1px solid #505050',
+                  }}
+                >
+                  +
+                </span>
+                &nbsp;&nbsp;
+                {/* do not localize "shapefile" */}
+                {t`Upload a`} shapefile
```

Comment:
I left "shapefile" out because I had noticed https://github.com/consbio/seedsource-ui/blob/0a492c9ef101aad191e2a6af0892a73ce7398f96/src/containers/ShapefileConstraint.tsx#L33
and https://github.com/consbio/seedsource-ui/blob/0a492c9ef101aad191e2a6af0892a73ce7398f96/src/config.ts#L165
Happy to put it back in, but seems suspicious : )